packages/flutter_tools/lib/src/build_system/targets/dart.dart

```diff
@@ -57,10 +99,31 @@ const Target kernelSnapshot = Target(
   name: 'kernel_snapshot',
   inputs: <Source>[
     Source.function(listDartSources), // <- every dart file under {PROJECT_DIR}/lib and .packages
+    Source.pattern('{CACHE_DIR}/artifacts/engine/common/flutter_patched_sdk/platform_strong.dill'),
   ],
   outputs: <Source>[
     Source.pattern('{BUILD_DIR}/main.app.dill'),
   ],
   dependencies: <Target>[],
-  invocation: compileKernel,
+  buildAction: compileKernel,
+);
+
+/// Generate an ELF binary from a dart snapshot.
+// TODO(jonahwilliams): does this need a dependency on gen_snapshot?.
+const Target aotElf = Target(
+  name: 'aot_elf',
+  inputs: <Source>[
+    Source.pattern('{BUILD_DIR}/main.app.dill'),
```

Comment:
Apropos of above: this should either list gen_snapshot, or there should be documentation with a whitelist of things that don't need to be listed, and gen_snapshot should be in that list.

Reply:
I added gen_snapshot and dart to this rule, and frontend_server and dart to the above